### DISCUSSION TITLE
How to use metric with several nuclei scan launch in parallel on the same machine ?

### DISCUSSION THREAD

**Discussion by bdoublet91** [projectdiscovery/nuclei#4013] [General]
### Nuclei version:
[INF] Nuclei Engine Version: v2.9.9

### Current Behavior:
When I want metric on scan, I start nuclei -m -mp 5000 but If I have several scan to monitor, I need to change port because one scan metric for one port, it's quite annoying. Do you have another way more scalable ?

### Expected Behavior:
A way to merge metrics in a file.
When I have 10 nuclei scan launched, I want to collect all nuclei metrics once, maybe in a file ?

### Steps To Reproduce:
One console: ./go/bin/nuclei -u https://example.com -m -mp 5000
Other console: ./go/bin/nuclei -u https://example.com -m -mp 5000

curl localhost:5000/metrics -> get only the first nuclei scan. It's ok because one service can be exposed to a port but it's difficult to collect metrics for several scan this way ...

Thanks

**Comment by dogancanbakir** [comment]
@bdoublet91 The current behavior you described in Nuclei is by design. Each scan operates independently to ensure accurate metrics. Unfortunately, Nuclei doesn't have a built-in mechanism to aggregate metrics from multiple scans. This is intentional to avoid inconsistencies and interferences between scans. As an interim solution, you could develop a script to collect metrics from each port separately and then aggregate them into a single file.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
